config/application.rb

```diff
@@ -261,6 +261,15 @@ def try_config_for(*args)
       config.three_scale.oauth2 = config_for(:oauth2)
     end
 
+    initializer :config_sts_client, after: :load_configs do
+      config.sts_client = CMS::STS::AssumeRoleWebIdentity.instance.tap do |sts_client|
```

Comment:
Feels a bit weird to initialize `AssumeRoleWebIdentity` here, without actually knowing whether it will be used or not - chances are it will not, as maybe IAM credentials will be used, so this singleton is there for nothing.
Not a big deal, but hmm...

Reply:
The idea in here is to have the ``Aws::STS::Client`` instance available throughout all application as part of the app's configuration. So, in here, we just create an object with the necessary configurations (maybe even missing something) and leave it in memory to be used or not.
Now, as a config, there shouldn't be any problem if we don't use it at all. Like how we don't use the `paperclip_defaults` config or even the `s3` one, in case we never hit the buckets to get/put assets (for CMS).

That was the line of thought 😬 Makes any sense?

Reply:
Yep, I get the idea, and that's what I don't like too much :upside_down_face: 
Indeed, we store config for S3, even though we might be using filesystem, and not S3.

That's the fact that it's actually an object instance, and not just some key-value hash that bugs me.

Reply:
Given `AssumeRoleWebIdentity` is a singleton, we can store the value as a `class_attrubute` (or a class-instance variable).

Reply:
>That's the fact that it's actually an object instance, and not just some key-value hash that bugs me.

I myself am not a big fan of globals but, for this case where we need to be sure this Client object never changes, so that the credentials can be fetch automatically, I think it makes sense. Specially since it's a Singleton.
For what it's worth, `Rails.application.config.sts_client` is pretty much a Hash (`.as_json` 😬) until credentials are requested, which'll only happen when STS is used over IAM.

Reply:
>Given AssumeRoleWebIdentity is a singleton, we can store the value as a class_attrubute (or a class-instance variable).

Looks good. Updated in c89d43e.

Reply:
Updated it again in e6d084e. Please check https://github.com/3scale/porta/pull/3066#discussion_r997180912.